File: AlgebraNArith/EMI/emi.cpp
#include <iostream>

#include <cmath>

#include <cstdlib>

using namespace std;

 

 

double   Get_emi( double principal , double rate , int number_of_year )

{

 

       double percent_rate = rate/(12*100);

 

       return  principal/(( 1 - pow(1+percent_rate,-number_of_year*12))/percent_rate);

 

}

 

 

void main(int argc , char **argv)

{

 

 

    if ( argc != 4 ) {

 

         fprintf(stdout,"Usage: EmiSched <principal> <rate> <period> \n");

         return;

 

    }

 

    double principal = atof(argv[1]);

    double rate      = atof(argv[2]);

    long   period    = atol(argv[3]);

    

    double emi =  Get_emi(principal,rate,period);

 

    double temp_principal = principal;

 

 

    long months = period*12;

 

    long curr_month = 0;

 

 

    double curpri =  principal; 

    while ( curr_month < months )

    {

          

          double interest = curpri*(rate/(12*100));

          fprintf(stdout,"%f\t%f\t%f\t%f\n",curpri,emi,emi-interest,interest);

          curpri -= (emi-interest); 

          curr_month++;

    }  

 

 

    return; 

 

 

 

}

File: AlgebraNArith/EMI/Emi.cs
using System;

namespace Test
{

public class Emi
{


  public static double Get_Emi( double principal , double rate , long n )
  {
     

       double percent_rate = rate/(12.0*100.0);

 

       return  principal/(( 1 - Math.Pow(1+percent_rate,-n*12.0))/percent_rate);

  }

 


 

  public static void Main(String [] args )
  {

        if ( args.Length != 3 )
        {

           Console.WriteLine("Emi  <Principal> <rate> <period> ");

           return;
        }

        double principal = 0;
        double rate   = 0;
        long n         = 0; 

        try {
           principal = Double.Parse(args[0]);
           rate      = Double.Parse(args[1]);
           n         = Int32.Parse(args[2]);
        }
        catch(Exception  ) {
         
          Console.WriteLine("Exception while parsing Command Line argument");
          Console.WriteLine("Emi  <Principal> <rate> <period> ");
          return;
        }


        
         
        double emi =  Get_Emi(principal,rate,n);




        double temp_principal = principal;
        long months = n*12;

 

        long curr_month = 0;
 

        double curpri =  principal; 
 
        while ( curr_month < months )

        {

          

          double interest = curpri*(rate/(12*100));

          Console.WriteLine(curpri.ToString() + " "+ emi.ToString() + " " + (emi-interest).ToString() + " " + interest.ToString() );    

          curpri -= (emi-interest); 

          curr_month++;

        }  

 

 

    return;   
        

  }




}

}
